### PR TITLE
Add fetch mixin functionality from the weather component

### PR DESCRIPTION
**Pull request by alex-deaconu**
Re-name `_tryGetCache` function to prevent override

### CODE REVIEW

**Comment by alex-deaconu** [comment]
@ezequielc please review. Thanks!

File: test/unit/fetch-mixin.html [hunk 1]
```diff
@@ -0,0 +1,310 @@
...
+
+          assert.isTrue( fetchMixin._requestRetryCount === 0 );
+          assert.isTrue( fetchMixin._refresh.called );
+          assert.isTrue( fetchMixin._refresh.calledWith(1000 * 60 * 10) );
+
+          assert.deepEqual( loggerMixin.log.getCall(0).args, ["error", "request error", {error: null}] );
+
+        });
+
+        test( "should raise 'request-error' on request failures", () => {
```
Comment (ezequielc) [inline test/unit/fetch-mixin.html]: @alex-deaconu The test message here is misleading. It doesn't send 'request-error', it calls the callback instead, right? Thanks

File: test/unit/fetch-mixin.html [hunk 2]
```diff
@@ -0,0 +1,310 @@
...
+        fetchMixin._getData().then(() => {
+          assert.isFalse( handleResponse.called );
+
+          assert.isTrue( fetchMixin._requestData.called );
+
+          done();
+        });
+      });
+
+      test( "should call API if cache is not available", done => {
```
Comment (ezequielc) [inline test/unit/fetch-mixin.html]: @alex-deaconu This test message is the same as the previous one. I think you should change it to "should call API if data is not cached". Thanks

File: test/unit/fetch-mixin.html [hunk 3]
```diff
@@ -0,0 +1,310 @@
...
+          clock.tick(5000);
+
+          assert.isTrue( fetchMixin._getData.calledOnce );
+        });
+
+      });
+
+    });
+
+    suite( "_processWeatherData", () => {
```
Comment (ezequielc) [inline test/unit/fetch-mixin.html]: @alex-deaconu I think this should be _getData(), right?

File: test/unit/fetch-mixin.html (hunk 1 above)
Comment (alex-deaconu) [reply test/unit/fetch-mixin.html]: @ezequielc you're right, the reason is they are tests that were modified from weather. I will review and correct. Thanks!

Comment (alex-deaconu) [reply test/unit/fetch-mixin.html]: @ezequielc I've updated the test descriptions as you suggested. Please see the last commit. Thanks!

**Comment by alex-deaconu** [comment]
@fjvallarino FYI this is the mixin/PR I was referring to. This is pretty much ready for merge so it can be used for RSS. Thanks.

File: test/unit/fetch-mixin.html (hunk 1 above)
Comment (ezequielc) [reply test/unit/fetch-mixin.html]: @alex-deaconu Looks good 👍

**Comment by fjvallarino** [comment]
@alex-deaconu thanks!!!

**Comment by ezequielc** [comment]
👍

**Pull request merged**